packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -388,6 +389,126 @@ class RelativePositionedTransition extends AnimatedWidget {
   }
 }
 
+/// Animates the different properties of a [Container].
+class ContainerTransition extends AnimatedWidget {
+  /// Creates a widget that animates painting, positioning, and sizing around
+  /// its child.
+  /// 
+  /// The `color` animation's value is a shorthand for
+  /// `decoration: new BoxDecoration(backgroundColor: color)`, which means you
+  /// cannot supply both a `color` and a `decoration` argument. If you want to
+  /// have both a `color` and a `decoration`, you can have your `decoration`
+  /// animation also control the `backgroundColor` of the `BoxDecoration`.
+  /// 
+  /// Also see [Container].
+  ContainerTransition({
+    Key key,
+    this.alignment,
+    this.padding,
+    this.color,
+    this.decoration,
+    this.foregroundDecoration,
+    this.height,
+    this.width,
+    this.constraints,
+    this.margin,
+    this.transform,
+    @required this.child,
+  }) : super(key: key, listenable: new Listenable.merge(<Listenable>[
```

Comment:
There's a more efficient way to implement this widget.  Rather than rebuilding the container each time any one of the animations tick, we could created "transition" versions of all the component widgets have have each one rebuild only when its corresponding animation ticks.  That would also let someone use one of the components separately.  For example, in your use case, I suspect you just want a DecoratedBoxTransition rather than spamming all the other properties too.